### DISCUSSION TITLE
Audacity API

### DISCUSSION THREAD

**Discussion by jacoby149** [audacity/audacity#2018] [Ideas & Feature Requests]
In the  preferences, let a user enable HTTP and configure Audacity to a port on their computer
Alternative to mod script piping, let Audacity be controlled via. an HTTP CRUD REST API
Send HTTP requests to Audacity, and Audacity performs the command corresponding to the sent HTTP request.

**Comment by SteveDaulton** [comment]
This proposal sounds potentially dangerous from a security point of  view.

**Reply by jacoby149** [reply]
definitely why it should only function upon enabling, similar to the mod script pipe feature

**Comment by LWinterberg** [comment]
I've seen something like this with [MakeHuman's Blender plugin](https://github.com/makehumancommunity/makehuman-plugin-for-blender), where instead of exporting a model from makehuman and then importing it into blender, it creates a websocket and automatically loads your stuff in. 

It's quite cool, but it also feels super weird to use, as if the program is doing a stage play and just kinda skipping over act 3. 

(And yes, this would require a very, very good look at networking security)

**Reply by jacoby149** [reply]
yeah that looks like a very cool plugin

**Comment by SteveDaulton** [comment]
Is there a use case for this proposal that cannot already be done with mod-script-pipe?
